packages/zosfiles/src/api/methods/create/Create.ts

```diff
@@ -134,6 +136,34 @@ export class Create {
         }
     }
 
+    public static async dataSetLike(session: AbstractSession, dataSetName: string, options: ICreateDataSetLikeOptions): Promise<IZosFilesResponse> {
+        const tempOptions = JSON.parse(JSON.stringify(options));
+
+        // Required
+        ImperativeExpect.toNotBeNullOrUndefined(dataSetName, ZosFilesMessages.missingDatasetName.message);
+
+        try {
+            const endpoint: string = ZosFilesConstants.RESOURCE + ZosFilesConstants.RES_DS_FILES + "/" + dataSetName;
+
+            const data = await ZosmfRestClient.postExpectString(session, endpoint, [], JSON.stringify(tempOptions));
+
+            let response = "";
+            if (!isNullOrUndefined(tempOptions.showAttributes)) {
+                delete tempOptions.showAttributes;
+                if (tempOptions.showAttributes) {
+                    response = TextUtils.prettyJson(List.dataSet(session, dataSetName, { attributes: true }));
```

Comment:
`List.dataSet` returns a Promise so must be awaited. Please also limit how much of the response is printed to `List.dataSet(...).apiResponse.items[0]`, so that it only displays the dataset attributes rather than an entire API response.

Reply:
Hello again @tjohnsonBCM :)

I am not able to return the same dataset attributes that are returned from function `Create.dataSet()`.

The dataset attributes returned by the API are [these attributes](https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.2.0/com.ibm.zos.v2r2.izua700/IZUHPINFO_API_RESTFILES_JSON_Documents.htm).

But the attributes returned by the `Create.dataSet()` function are [these attributes](https://www.ibm.com/support/knowledgecenter/SSLTBW_2.4.0/com.ibm.zos.v2r4.izua700/IZUHPINFO_API_CreateDataSet.htm).

`Create.dataSet()` just returns the attributes it has been passed as YAML, if `{ attributes = true }`. Since `Create.dataSetLike()` doesn't receive any attributes (it is just receiving the name of the dataset to copy attributes from) I am not sure how to retrieve the attributes available at the second link.

I'd like to do this so that the return value from `Create.dataSetLike()` is exactly the same as that from `Create.dataSet()`.

For now, `Create.dataSetLike()` is returning a selection of relevant attributes listed at the first link.

@tjohnsonBCM if you know how to use the API to get the attributes listed at the second link, please let me know! I looked at the API endpoints [here](https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.2.0/com.ibm.zos.v2r2.izua700/IZUHPINFO_API_RESTFILES.htm) but couldn't find anything that would work.

Reply:
@katelynienaber You're right, I don't think it's possible to use the API to get the attributes listed at the second link.

Discussed this at Zowe CLI Squad meeting. Either of the following approaches are fine:
* Keep the current behavior in this PR of `Create.dataSetLike` returning the list of attributes provided by z/OSMF list API.

  It's OK that this differs from the list of attributes returned by `Create.dataSet`. If we wanted to make all attribute lists consistent, perhaps `Create.dataSet` output should be changed. But that would be a breaking change that can be considered in the future.

  If this approach is taken, we need to be careful about calling multiple z/OSMF APIs in one API method. In the case where the z/OSMF create API succeeds and the z/OSMF list API fails, we should make it clear to the user that the dataset was created successfully. Most API methods call only one z/OSMF API so they don't have to handle this scenario.

  To address this, it would be good to add a unit test that validates if the create API call succeeds and the list API call throws an error, the command should still output a success message.
* Remove the `showAttributes` option from "Create.dataSetLike". It's a nice to have, but if it's too much effort to add now and handle the above scenario, it can be done in the future per community request.

Reply:
Hello, thanks for your review!

I will probably remove the show attributes parameter, but I am asking our team first.

Maybe they can think of a use case for returning them.